### PR TITLE
[#3] [UI] As a user, I can upload a CSV file containing keywords which will be stored

### LINKED ISSUE #3: [UI] As a user, I can upload a CSV file containing keywords which will be stored
- Only .csv file extentions can be selected for upload

### PR DESCRIPTION
**Pull request by liamstevens111**
- #3 

## What happened 👀

Allow users to upload a .csv file containing a list of keywords.

## Insight 📝

Handles the client side functionality for uploading of files. 

#18 Server side validation, parsing and resulting keyword storage will be handled in the Pull Request that follows.

## Proof Of Work 📹

Screenshot 2565-05-11 at 08 59 03

Screenshot 2565-05-11 at 08 59 29

### CODE REVIEW

File: test/google_search_data_viewer_web/controllers/keyword_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      user = insert(:user)
+
+      conn =
+        conn
+        |> init_test_session(user_id: user.id)
+        |> post("/keywords/upload", %{:file => file})
+
+      assert redirected_to(conn, 302) =~ "/keywords"
+      assert get_flash(conn, :info) =~ "File successfully uploaded"
+    end
```
Comment (byhbt) [inline test/google_search_data_viewer_web/controllers/keyword_controller_test.exs:48]: Here you can add one more test for uploading by not logged in user.

File: lib/google_search_data_viewer_web/templates/keyword/form.html.heex [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<.form id="keyword-upload-form" let={f} for={@conn} action={@action} multipart={true}>
+  <%= label f, :File %>
+  <%= file_input f, :file, accept: ".csv", required: true %>
+  <%= error_tag f, :file %>
+
+  <div>
+    <%= submit "Upload" %>
+  </div>
+</.form>
+
```
Comment (byhbt) [inline lib/google_search_data_viewer_web/templates/keyword/form.html.heex]: Please remove this empty line

File: test/google_search_data_viewer_web/plugs/ensure_authenticated_test.exs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      conn =
+        conn
+        |> init_test_session(%{})
+        |> fetch_flash()
+        |> EnsureAuthenticatedPlug.call([])
+
+      assert conn.assigns.current_user == nil
+      assert get_flash(conn, :error) =~ "Please sign in to use this service"
+      assert redirected_to(conn, 302) =~ "/"
+      assert conn.halted == true
```
Comment (byhbt) [inline test/google_search_data_viewer_web/plugs/ensure_authenticated_test.exs]:
```suggestion
      assert conn.assigns.current_user == nil
      assert conn.halted == true
      assert get_flash(conn, :error) =~ "Please sign in to use this service"
      assert redirected_to(conn, 302) =~ "/"
```

File: test/google_search_data_viewer_web/controllers/keyword_controller_test.exs (hunk 1 above)
Comment (liamstevens111) [reply test/google_search_data_viewer_web/controllers/keyword_controller_test.exs:48]: > 

Ok, will do. I initially thought the tests in the `ensure_authenticated` plug test will suffice but I will add that here too :)

**Review by byhbt — APPROVED** [review]

**Review by hanam1ni — APPROVED** [review]
Looking good 👍

**Pull request merged**